pkg/distro/sort.go

```diff
@@ -0,0 +1,61 @@
+package distro
+
+import (
+	"errors"
+	"regexp"
+	"sort"
+
+	"github.com/hashicorp/go-version"
+)
+
+var nameVerRe = regexp.MustCompile(`-[0-9]+`)
+
+// SortNames sorts the given list of distro names by name, version
+// taking version semantics into account (i.e. sorting 8.1 lower then
+// 8.10). Full semantic versioning is supported (see semver.org).
+//
+// Invalid version numbers will create errors but the sorting continue
+// and invalid numbers are sorted lower than anything else (so the
+// result is still usable in a {G,T}UI).
+func SortNames(distros []string) error {
+	var errs []error
+
+	sort.Slice(distros, func(i, j int) bool {
+		var name1, ver1, name2, ver2 string
+
+		nameVer1 := distros[i]
+		nameVer2 := distros[j]
+		sep1 := nameVerRe.FindStringIndex(nameVer1)
```

Comment:
@thozza Thanks, I like this idea. Reusing is always nicer.Doing the naive thing of just importing distroid into sort.go does not work because distro/distroid would have circular imports. So I went with a still naive but slightly but slightly less creation of a new distrosort package and added id.Version() which currently must return also return error because we expose Major/Mintor so there is no guarantee that the user may not just modify the version after ParseID. It's all in https://github.com/osbuild/images/pull/1012 and if you prefer this I'm happy to close this one. The other will not support things like "-beta" or other semver things but that is no regression and arguably those feature should come via distroid so that that it's supported everywhere not just in the sorting.